### DISCUSSION TITLE
用docker 编译过程中发生错误

### DISCUSSION THREAD

**Discussion by zpeak** [apache/doris#19316] [A - General / Q&A]
环境 centOs,   
Docker 拉的最新编译环境,
代码为apache-doris-1.2.4.1-src

错误提示为:
/root/apache-doris-1.2.4.1-src/be/src/vec/core/field.h: In member function 'doris::Status doris::ScrollParser::fill_columns(const doris::TupleDescriptor*, std::vector<COW<doris::vectorized::IColumn>::mutable_ptr<doris::vectorized::IColumn> >&, doris::MemPool*, bool*, const std::map<std::__cxx11::basic_string<char>, std::__cxx11::basic_string<char> >&)':
/root/apache-doris-1.2.4.1-src/be/src/vec/core/field.h:633:9: error: 'val' may be used uninitialized in this function [-Werror=maybe-uninitialized]
  633 |         new (&storage) StorageType(std::forward<T>(x));
      |         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/root/apache-doris-1.2.4.1-src/be/src/exec/es/es_scroll_parser.cpp:627:30: note: 'val' was declared here
  627 |                     __int128 val;
      |                              [^~~](url)

**Comment by zy-kkk** [comment]
If your compiled code is the Doris 1.2 branch, please use the 1.2 version of the docker development image: apache/doris:build-env-for-1.2
